pkg/cmd/release/create/create.go

```diff
@@ -266,17 +271,6 @@ func createRun(opts *CreateOptions) error {
 					opts.TagName, ghrepo.FullName(baseRepo))
 			}
 		}
-
-		if opts.NotesFromTag {
```

Comment:
Can you say a little bit more about why this is unnecessary? It seems like if I say `--notes-from-tag` and there is no tag with the name I provided., then I would expect that to be an error. In particular, it doesn't even fall back to the interactive flow because the `--notes-from-tag` flag was provided.

What am I missing?

Reply:
Ah, I suspect it is because there is a separate `--verify-tag` flag?

Reply:
This is unnecessary because it is the same check that is being done on line 260 above. The check above makes this one redundant. I can't think of a case where this did not result in duplicate calls to check if the remote tag exists.

Reply:
I don't think that's true empirically, because when I ran `gh release create --notes-from-tag <non-existent-tag>` it completed.

I don't have time to confirm now but I think the case is "there is no local tag". I suspect on https://github.com/cli/cli/blob/391ff1bd5ae00080790aade47a723a325f61877a/pkg/cmd/release/create/create.go#L264 that the `tagDescription` is an empty string (perhaps because we throw away the error check).

Reply:
In the case of no local tag and the `--notes-from-tag` flag specified I assumed we wanted a result of a successful creation of the release with an empty release body as there were no notes in the tag. Perhaps that is the wrong assumption though.

Reply:
Can you think of a good reason that someone would provide the tag and not expect to have notes generated from a tag? It seems more likely to me that someone would made a mistake in the provided tag and they'd want it to error out.

Reply:
Well the I can think of a few automated scripting cases where the person running `gh release create` was not the one who created the tag and thus does not know if it is an annotated tag or not and uses `--notes-from-tag` as a precautionary measure. Not sure that would be common. 

I don't feel strongly about this though. I can change it to error out if `--notes-from-tag` is specified and the local tag does not exist or is not an annotated tag.

Reply:
@andyfeller tie breaker!

Reply:
> In the case of no local tag and the `--notes-from-tag` flag specified I assumed we wanted a result of a successful creation of the release with an empty release body as there were no notes in the tag. Perhaps that is the wrong assumption though.

**What local repository states are we considering when creating a release?**

If there is no local tag because the repository hasn't pulled changes from remote but a remote tag exists, then should the local repository fetch changes and then continue on or should we read information from the remote?

> Can you think of a good reason that someone would provide the tag and not expect to have notes generated from a tag? It seems more likely to me that someone would made a mistake in the provided tag and they'd want it to error out.

If you're asking whether we should ensure the tag provided contains a description or not, we probably should expose a flag that allows someone to determine whether this is the case or not.  I imagine there are use cases on both sides and probably easier to have a default behavior _(checking that some notes are generated)_ that people can override.

Reply:
> If you're asking whether we should ensure the tag provided contains a description or not

No, I'm asking whether we should ensure the tag exists at all locally, which was in the original PR but is removed by this diff.

Reply:
@williammartin I addressed this with my latest commit if you wouldn't mind taking another look 🙇

Reply:
```
➜  cli git:(cleanup-release-create) ./bin/gh release create --notes-from-tag v1.2.8
cannot generate release notes from tag v1.2.8 as it does not exist locally
```

LGTM